.github/workflows/build_and_test.yml

```diff
@@ -123,6 +123,19 @@ jobs:
     - name: Check labeler configuration
       run: scripts/check-labeler.pl .github/labeler.yml
 
+
+    # NOTE: this is basically an inline of the official, public gh action
+    # (https://github.com/ast-grep/action).
```

Comment:
The official action is a full-on JavaScript wrapper around `npm install ast-grep && ast-grep scan [...]`. With all the noise about github actions and needing to audit/lock things down, it makes more sense to just maintain these <10 lines of code ourselves.